### ISSUE TITLE
Run WhichCells for multiple genes

### ISSUE THREAD

**Issue by fly4all** [satijalab/seurat#2013]
I wanted to check number of cells expressing certain genes in specific clusters.
Since there are a lot of genes I need to do this for (~1000), I tried to do it in a for loop.
However, the problem I'm having is that WhichCells does not take a variable as an input in "WhichCells(...expression = geneName >= 0.5...)". Can you suggest an alternative or any way around this?

Here's the code I'm currently trying:
```
genes = c("dpr","dpr2","dpr3","dpr4","dpr5")
clusters = c(1,7,13,5,9,2,3,4)
x = matrix(,nrow=length(genes),ncol=(length(clusters)+1))
for (g in 1:length(genes)){
  for (c in 2:(length(clusters)+1)){
    nCells <- length(WhichCells(object = decoder, idents= c, expression = genes[g] >= 0.5))
    x[g,1] <- genes[g]
    x[g,c] <- nCells 
  }

```
the nCells line here doesn't work because it thinks genes[g] is an actual gene name

Thank you!

**Comment by mojaveazure** [comment]
Answered in https://github.com/satijalab/seurat/issues/1839#issuecomment-512560781

**Comment by NMathias8** [comment]
I have been having this same issue when trying to run functions like WhichCells where the feature name is not in quotes, but haven't found an issue in the Seurat help threads. While searching for a solution to this issue with add_column(parameter = value) I found it given here https://stackoverflow.com/questions/45741498/add-column-in-tibble-with-variable-column-name

which linked to https://www.tidyverse.org/blog/2019/06/rlang-0-4-0/

tldr
The solution here is to enclose your variable with double curly brackets
nCells <- length(WhichCells(object = decoder, idents= c, expression = {{genes[g]}} >= 0.5))

If using inside a function as a parameter like add_column(),
 := take the place of =

var <- "Gene1"
dataframe <- dataframe %>% add_column( {{var}} := value)
